### PR TITLE
EDU-3194: Adds next retry delay explanation to Python docs

### PR DESCRIPTION
**Pull request by fairlydurable**
- Core explanation wording taken from Drew H
- Sample adapted from Chad
- Transition material added between intro and sample, along lines of other SDKs, to introduce sample

Will need technical review for accuracy

### CODE REVIEW

File: docs/develop/python/failure-detection.mdx [hunk 1]
```diff
@@ -180,6 +180,30 @@ from temporalio.common import RetryPolicy
...
+```python
+from temporalio.exceptions import ApplicationError
+from datetime import timedelta
+
+@activity.defn
+async def my_activity(attempt):
+    # Your activity logic goes here
+    ...
+    raise ApplicationError(
+        "Error encountered on attempt " + str(attempt),
```
Comment (dandavison) [inline docs/develop/python/failure-detection.mdx:202]:
```suggestion
        f"Error encountered on attempt {attempt}",
```

Comment (fairlydurable) [reply docs/develop/python/failure-detection.mdx:202]: Thank you. Adopted.

**Review by dandavison — APPROVED** [review]
LGTM with the code suggestion

File: docs/develop/python/failure-detection.mdx [hunk 2]
```diff
@@ -180,6 +180,30 @@ from temporalio.common import RetryPolicy
...
+from datetime import timedelta
+
+@activity.defn
+async def my_activity(attempt):
+    # Your activity logic goes here
+    ...
+    raise ApplicationError(
+        f"Error encountered on attempt {attempt}",
+        next_retry_delay=timedelta(seconds=3 * attempt),
+    )
```
[the remark this answers is not in the document]
Comment (fairlydurable) [reply docs/develop/python/failure-detection.mdx]: Adopted and merged in. Grabbed from the slack thread, so marking this as resolved without committing suggestion.

File: docs/develop/python/failure-detection.mdx [hunk 3]
```diff
@@ -180,6 +180,32 @@ from temporalio.common import RetryPolicy
         )
 ```
 
+### Override the retry interval with `next_retry_delay` {#next-retry-delay}
+
+To override the next retry interval set by the current policy, pass `next_retry_delay` when raising an [ApplicationError](/references/failures#application-failure) in an Activity.
+This value replaces and overrides whatever the retry interval would normally be on the retry policy.
+
+For example, you can set the delay interval based on an Activity's attempt count.
+In the example below, the retry delay starts at 3 seconds after the first attempt.
```
Comment (napcs) [inline docs/develop/python/failure-detection.mdx:189]:
```suggestion
In the following example, the retry delay starts at 3 seconds after the first attempt.
```

**Review by napcs — APPROVED** [review]

**Pull request merged**